### PR TITLE
Add pitch range control to preprocess and add postprocess

### PR DESCRIPTION
**Pull request by Sebastian-Larsson**
Postprocess adds the ability to save a npy file to midi.

### CODE REVIEW

File: processing/npy_to_midi_example.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
+from cgi import test
```
Comment (LeifSulaiman) [inline processing/npy_to_midi_example.py:1]: What is this import used for

File: processing/postprocess.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    categories = ['Piano', 'Chromatic Percussion', 'Organ',
+                  'Guitar', 'Bass', 'Strings', 'Ensemble',
+                  'Brass', 'Reed', 'Pipe', 'Synth Lead',
+                  'Synth Pad', 'Synth Effects', 'Ethnic']
+
+    if song_type not in VALID_TYPES:
+        raise ValueError("song_type must be one of %r" %VALID_TYPES)
+
+    if song.is_cuda: # Make sure song Tensor is on cpu
+        song = song.cpu()
```
Comment (LeifSulaiman) [inline processing/postprocess.py:39]: I prefer a new line after 'if' statement.

File: processing/postprocess.py [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    print(song[:-1].shape)
```
Comment (LeifSulaiman) [inline processing/postprocess.py:40]: Stray prints, remove this and line 42.

File: processing/postprocess.py [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+    song = pad_song(song)
+    print(song[:-1].shape)
+    merged_track = pypianoroll.Multitrack(tempo=np.array([bpm], dtype=float))
```
Comment (LeifSulaiman) [inline processing/postprocess.py:43]: Why is the resolution not included? Same for line 46.

File: processing/postprocess.py [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+def pad_song(song : Tensor):
+    """Adds padding to pytorch Tensor.
+
+    Pads input Tensor to have the shape [<LENGTH>, 128]. Midi files need to have height 128 because thats the
+    number of possible notes. Having a lower amount will not save the file correctly and notes might get
+    shifted.
+
+    Parameters
+    ----------
+    track : Tensor
```
Comment (LeifSulaiman) [inline processing/postprocess.py:66]: Parameter is called song, not track.

File: processing/preprocess.py [hunk 6]
```diff
@@ -1,4 +1,5 @@
 import os
+from turtle import up
```
Comment (LeifSulaiman) [inline processing/preprocess.py:2]: Unused import, remove.

File: processing/preprocess.py [hunk 7]
```diff
@@ -25,23 +26,38 @@ class Preprocess():
         Target path for processed files.
     resolution : int = 8
         Sets the resolution of the pianoroll (i.e. sampling rate).
+    pitch_range : set = (0, 127)
+        If user selects values, the read midi will have a reduced range.
```
Comment (LeifSulaiman) [inline processing/preprocess.py:30]: Specify that user must choose range of (>=0, <=127).

**Review by LeifSulaiman — CHANGES_REQUESTED** [review]
Review changes and add comments in files:

'processing/npy_to_midi__example.py',
'processing/postprocess.py',
'processing/preprocess.py'

File: processing/preprocess.py (hunk 6 above)
Comment (Sebastian-Larsson) [reply processing/preprocess.py:2]: Done, also removed `pretty_midi as pm` and `json` as they were both unused.

File: processing/postprocess.py (hunk 4 above)
Comment (Sebastian-Larsson) [reply processing/postprocess.py:43]: Included in multitrack. Standardtrack doesn't need resolution.

**Review by LeifSulaiman — APPROVED** [review]
Approve changes.

**Pull request merged**